crates/biome_graphql_parser/src/parser/definitions/operation.rs

```diff
@@ -69,69 +62,213 @@ impl ParseRecovery for SelectionListParseRecovery {
     }
 }
 
-pub(crate) struct SelectionList;
+#[derive(Default)]
+struct VariableDefinitionList;
 
-impl SelectionList {
-    pub(crate) fn new() -> Self {
-        Self
-    }
-}
-
-impl ParseNodeList for SelectionList {
+impl ParseNodeList for VariableDefinitionList {
     type Kind = GraphqlSyntaxKind;
     type Parser<'source> = GraphqlParser<'source>;
 
-    const LIST_KIND: Self::Kind = GRAPHQL_SELECTION_LIST;
+    const LIST_KIND: Self::Kind = GRAPHQL_VARIABLE_DEFINITION_LIST;
 
     fn parse_element(&mut self, p: &mut Self::Parser<'_>) -> ParsedSyntax {
-        parse_selection(p)
+        parse_variable_definition(p)
     }
 
     fn is_at_list_end(&self, p: &mut Self::Parser<'_>) -> bool {
-        p.at(T!['}'])
+        p.at(T![')'])
     }
 
     fn recover(
         &mut self,
         p: &mut Self::Parser<'_>,
         parsed_element: ParsedSyntax,
     ) -> biome_parser::parse_recovery::RecoveryResult {
-        parsed_element.or_recover(p, &SelectionListParseRecovery, expected_any_selection)
+        parsed_element.or_recover(
+            p,
+            &VariableDefinitionListParseRecovery,
+            expected_variable_definition,
+        )
     }
 }
 
+struct VariableDefinitionListParseRecovery;
+
+impl ParseRecovery for VariableDefinitionListParseRecovery {
+    type Kind = GraphqlSyntaxKind;
+    type Parser<'source> = GraphqlParser<'source>;
+    const RECOVERED_KIND: Self::Kind = GRAPHQL_VARIABLE_DEFINITION;
+
+    fn is_at_recovered(&self, p: &mut Self::Parser<'_>) -> bool {
+        is_at_variable(p)
+    }
+}
+
+/// https://spec.graphql.org/October2021/#sec-Language.Operations.Query-shorthand
+#[inline]
+pub(crate) fn parse_operation_definition(p: &mut GraphqlParser) -> ParsedSyntax {
+    if !is_at_operation(p) {
+        return Absent;
+    }
+
+    if !p.at_ts(OPERATION_TYPE) {
```

Comment:
This reads a bit confusingly if we don't know what's inside `is_at_operation()`. By changing this into a positive test, we know exactly what we are testing for.

```suggestion
    if is_at_selection_set(p) {
```